From fe71fc36f531907d7d1909e7e33503eb081d3d08 Mon Sep 17 00:00:00 2001
From: shamim ali <ashamim@ami.com>
Date: Mon, 28 Apr 2025 05:50:38 -0400
Subject: [PATCH] Added errorFalg and infoFlags for multiple logTypes

This patch added for infoFlags and errorFlags for different logType
supported for logging objects

Signed-off-by: shamim ali <ashamim@ami.com>
---
 include/defaults.hpp  |  6 ++++++
 include/interface.hpp | 13 +++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index ae99a7c..0c2cb7f 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -417,6 +417,12 @@ setting->addProperty("RetryIntervalMS", static_cast<uint8_t>(20));
 
     setting->addProperty("QuiesceOnHwError", false);
     setting->addProperty("SelPolicy", "xyz.openbmc_project.Logging.Settings.Policy.Circular");
+    setting->addProperty("InfoFlags", std::vector<std::pair<std::string, bool>>{
+	    {"default", false},{"ipmi", false},{"raid", false}});
+
+    setting->addProperty("ErrorFlags", std::vector<std::pair<std::string, bool>>{
+	    {"default", false},{"ipmi", false},{"raid", false}});
+
 
     setting = &settings.emplace_back(objectServer,
                                      "/xyz/openbmc_project/pfr/last_events",
diff --git a/include/interface.hpp b/include/interface.hpp
index d36d815..2f3e67e 100644
--- a/include/interface.hpp
+++ b/include/interface.hpp
@@ -112,6 +112,19 @@ struct SettingsInterface
             }
             return resp;
         }
+	else if constexpr (std::is_same_v<T, std::pair<std::string, bool>>)
+        {
+            if (data.is_array() && data.size() == 2)
+            {
+                const std::string* key = data[0].get_ptr<const std::string*>();
+                const bool* val = data[1].get_ptr<const bool*>();
+                if (key && val)
+                {
+                    resp = std::make_pair(*key, *val);
+                }
+            }
+            return resp;
+        }
         else
         {
             static_assert(!std::is_same_v<T, T>, "Unsupported Type");
-- 
2.34.1

